include/boost/geometry/strategies/cartesian/point_in_box.hpp

```diff
@@ -53,9 +56,75 @@ struct covered_by_range
 };
 
 
+template <typename Geometry>
+struct within_range<Geometry, 0, spherical_tag>
+{
+    template <typename Value1, typename Value2>
+    static inline bool apply(Value1 const& value, Value2 const& min_value, Value2 const& max_value)
+    {
+        typedef typename select_most_precise
+            <
+                Value1, Value2
+            >::type calc_t;
+        typedef typename coordinate_system<Geometry>::type::units units_t;
+        typedef math::detail::constants_on_spheroid<calc_t, units_t> constants;
+
+        // min <= max <=> diff >= 0
+        calc_t const diff_ing = max_value - min_value;
+
+        // if containing covers the whole globe it contains all
+        if (!math::smaller(diff_ing, constants::period())) // >= period
+            return true;
+
+        // calculate positive longitude translation with bing_min as origin
+        calc_t const c0 = 0;
+        calc_t diff_min = value - min_value;
+        math::normalize_longitude<units_t, calc_t>(diff_min);
+        if (diff_min < c0) // [-180, 180] -> [0, 360]
+            diff_min += constants::period();
+
+        return diff_min > c0 && min_value + diff_min < max_value;
+    }
+};
+
+
+template <typename Geometry>
+struct covered_by_range<Geometry, 0, spherical_tag>
+{
+    template <typename Value1, typename Value2>
+    static inline bool apply(Value1 const& value, Value2 const& min_value, Value2 const& max_value)
+    {
+        typedef typename select_most_precise
+            <
+                Value1, Value2
+            >::type calc_t;
+        typedef typename coordinate_system<Geometry>::type::units units_t;
+        typedef math::detail::constants_on_spheroid<calc_t, units_t> constants;
+
+        // min <= max <=> diff >= 0
+        calc_t const diff_ing = max_value - min_value;
+
+        // if containing covers the whole globe it contains all
+        if (!math::smaller(diff_ing, constants::period())) // >= period
+            return true;
+
+        // calculate positive longitude translation with min_value as origin
+        calc_t const c0 = 0;
+        calc_t diff_min = value - min_value;
+        math::normalize_longitude<units_t, calc_t>(diff_min);
+        if (diff_min < c0) // [-180, 180] -> [0, 360]
+            diff_min += constants::period();
+
+        return min_value + diff_min <= max_value;
+    }
+};
```

Comment:
So this code is duplicated, the only difference apparently being the last line? It must be possible to share more functionality?